tools/wptrunner/wptrunner/executors/protocol.py

```diff
@@ -70,9 +70,10 @@ def setup(self, runner):
             msg = "Post-connection steps failed"
             self.after_connect()
         except Exception:
-            if msg is not None:
-                self.logger.warning(msg)
-            self.logger.warning(traceback.format_exc())
+            message = "Protocol.setup caught an exception:\n"
+            message += f"{msg}\n" if msg is not None else ""
+            message += traceback.format_exc()
+            self.logger.warning(message)
             raise
```

Comment:
Converting this to `raise from e` where `e` is the exception object is also possible now, and might help?